Reduce cloudfare worker size

Reduce the size of worker as the cloudfare deploy is failing:
```
✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/gei-staging/versions) failed.

  Your Worker exceeded the size limit of 3 MiB. Please upgrade to a paid plan to deploy Workers up to 10 MiB. https://dash.cloudflare.com/***/workers/plans [code: 10027]
  To learn more about this error, visit: https://developers.cloudflare.com/workers/platform/limits/#worker-size

  
  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose


🪵  Logs were written to "/home/runner/.config/.wrangler/logs/wrangler-2026-04-27_07-56-13_299.log"

 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

Also, add some CI test to check that and update the documentation.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -73,11 +73,22 @@ jobs:
       - uses: actions/setup-node@v4
         with: { node-version: 20, cache: 'pnpm' }
       - run: pnpm install --frozen-lockfile
-      - name: Build
+      - name: Build (OpenNext)
         env:
           NEXT_PUBLIC_SUPABASE_URL: http://localhost:54321
           NEXT_PUBLIC_SUPABASE_ANON_KEY: dummy
-        run: pnpm build
+        run: pnpm opennextjs-cloudflare build
+      - name: Check Worker Size
+        run: |
+          # Cloudflare free plan limit is 3MB (gzipped)
+          # We check the handler size as it's the main contributor
+          SIZE=$(gzip -c .open-next/server-functions/default/handler.mjs | wc -c)
+          LIMIT=3145728 # 3 MiB
+          echo "Worker handler gzipped size: $SIZE bytes"
+          if [ "$SIZE" -gt "$LIMIT" ]; then
+            echo "Error: Worker size exceeds 3MB limit!"
+            exit 1
+          fi
 
   database-tests:
     runs-on: ubuntu-latest

--- a/docs/architecture.md
+++ b/docs/architecture.md
@@ -95,6 +95,15 @@ for the full repo map. Key directories:
 - CI (`.github/workflows/ci.yml`) runs lint/typecheck/unit on every PR,
   RLS tests against a local Supabase spin-up, and Playwright e2e on PRs
   to `main`.
+- **Worker Size Monitoring:** The CI build job verifies that the gzipped size of the Cloudflare Worker handler stays under the 3 MiB limit imposed by the Cloudflare Workers Free plan.
+
+## Deployment & Worker Limits
+
+- **Platform:** Cloudflare Workers via OpenNext.
+- **Limits:** 3 MiB gzipped bundle size (Free plan).
+- **Optimizations:**
+  - **Dynamic Imports:** Large libraries like `exceljs` are dynamically imported within their respective utility functions to move them out of the main server bundle.
+  - **Package Import Optimization:** `next.config.ts` uses `optimizePackageImports` for `lucide-react` to ensure efficient tree-shaking of icons.
 
 ## Phases
 

--- a/lib/exporters/xlsx.ts
+++ b/lib/exporters/xlsx.ts
@@ -1,5 +1,3 @@
-import ExcelJS from 'exceljs';
-
 export type XlsxColumn<T> = {
   key: keyof T;
   header: string;
@@ -23,6 +21,7 @@ export async function buildXlsx<T>({
   columns,
   rows,
 }: XlsxInput<T>): Promise<ArrayBuffer> {
+  const ExcelJS = (await import('exceljs')).default;
   const wb = new ExcelJS.Workbook();
   const ws = wb.addWorksheet(sheetName);
 

--- a/next.config.ts
+++ b/next.config.ts
@@ -10,6 +10,9 @@ const nextConfig: NextConfig = {
   // prints so both the owner and anyone on the same Wi-Fi can load the
   // app during development. Production (`next start`) ignores this list.
   allowedDevOrigins: ['localhost', '127.0.0.1', '10.0.6.19'],
+  experimental: {
+    optimizePackageImports: ['lucide-react'],
+  },
 };
 
 export default nextConfig;

